src/robot/libraries/Process.py

```diff
@@ -524,8 +526,37 @@ def _manage_process_timeout(self, handle, on_timeout):
 
     def _wait(self, process):
         result = self._results[process]
-        result.rc = process.wait() or 0
-        result.close_streams()
+
+        _take_stdout = (process.stdout and (not process.stdout.closed))
+        _take_stderr = (process.stderr and (not process.stderr.closed))
+
+        # This is needed to handle stdin closes identically to how it was when the
+        # wait function was used.
+        _stdin = process.stdin
+        process.stdin = process.stdin if (process.stdin and not process.stdin.closed) else io.StringIO("")
+        while True:
+            with contextlib.suppress(subprocess.TimeoutExpired):
+                # this is to allow the process to be aborted when on windows
+                # an exception is used to mange the test timeout situation
+                (_stdout, _stderr,) = process.communicate(timeout=0.1)
+                break
+
+        process.stdin = _stdin
+
+        result.rc = process.returncode
+
+        # This is needed to handle stdin closes identically to how it was when the
+        # wait function was used.
+        if _stdout:
+            result._stdout = result._format_output(_stdout if _take_stdout else "")
+        
+        if _stderr:
+            result._stderr = result._format_output(_stderr if _take_stderr else "")
+
+        for f in [process.stdout, process.stderr] + result._custom_streams:
+            with contextlib.suppress(AttributeError):
+                f.close()
```

Comment:
Accessing `_private` members of the `result` object like the above code does is not ok. Making `result.stdout` property settable is a better solution.